config/config.go

```diff
@@ -0,0 +1,144 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package config defines configurable values shared between Source and Destination
+// and implements a method to parse them.
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/conduitio-labs/conduit-connector-mongo/validator"
+)
+
+// AuthMechanism defines a MongoDB authentication mechanism.
+type AuthMechanism string
+
+// The list of available authentication mechanisms is listed below.
+const (
+	SCRAMSHA256AuthMechanism AuthMechanism = "SCRAM-SHA-256"
+	SCRAMSHA1AuthMechanism   AuthMechanism = "SCRAM-SHA-1"
+	MongoDBCRAuthMechanism   AuthMechanism = "MONGODB-CR"
+	MongoDBAWSAuthMechanism  AuthMechanism = "MONGODB-AWS"
+	X509AuthMechanism        AuthMechanism = "X.509"
+)
+
+// authMechanismMap holds a map with all possible auth mechanisms.
+// It's a helper for checking if a string is a valid auth mechanism.
+var authMechanismMap = map[string]AuthMechanism{
```

Comment:
Just asking. 
Would it be better if you use the next map: `map[string]struct{}`?

And then just check it out like that:

```
if _, ok := authMechanismMap[strings.ToUpper(authMechanismStr)]; ok {
		return strings.ToUpper(authMechanismStr), nil
}
```

Reply:
Or in the case of `map[string]bool`, we could just do the following:
```go
if authMechanismMap[strings.ToUpper(authMechanismStr)] {
    return AuthMechanism(strings.ToUpper(authMechanismStr)), nil
}
```

Hmm, but actually we have a pretty small number of possible values for the `AuthMechanism`, so maybe we can just do:
```go
// IsValid checks if the underlying AuthMechanism is valid.
func (am AuthMechanism) IsValid() bool {
	switch am {
	case SCRAMSHA256AuthMechanism, SCRAMSHA1AuthMechanism, MongoDBCRAuthMechanism,
		MongoDBAWSAuthMechanism, X509AuthMechanism:
		return true
	}

	return false
}
```

And in the `Parse` function we'll check:
```go
// validate auth mechanism if it's not empty
if config.Auth.Mechanism != "" && !config.Auth.Mechanism.IsValid() {
	return Config{}, &InvalidAuthMechanismError{
		AuthMechanism: config.Auth.Mechanism,
	}
}
```

What do you think?

Reply:
The last one is pretty damn good :)

And just a small question. Do we actually need the first check in the condition?
```config.Auth.Mechanism != ""```

Reply:
The `config.Auth.Mechanism` can be empty. In this case MongoDB will use the default authentication mechanism depending on a MongoDB server version.

Reply:
Oh yeah, right 👍